program/Cargo.toml

```diff
@@ -12,14 +12,15 @@ no-entrypoint = []
 test-sbf = []
 
 [dependencies]
+arrayvec = "0.7.6"
 bitflags = { version = "2.6.0", features = ["serde"] }
 bytemuck = { version = "1.21.0", features = ["derive"] }
 num_enum = "0.7.3"
-generic-array = { version = "1.1.1", features = ["serde"], default-features = false }
 bincode = "1.3.3"
 num-derive = "0.4"
 num-traits = "0.2"
 solana-program = "2.1.0"
+solana-signature = "2.1.0"
```

Comment:
Ended up using this just for convenience of constants. We still store as a `&'a [u8; SIGNATURE_BYTES]` to avoid the alloc/copy in the shred / context